support/cas-server-support-actions/src/main/java/org/apereo/cas/web/flow/login/mfa/MfaInitializeAction.java

```diff
@@ -0,0 +1,35 @@
+package org.apereo.cas.web.flow.login.mfa;
+
+import org.apereo.cas.services.MultifactorAuthenticationProvider;
+import org.apereo.cas.util.spring.ApplicationContextProvider;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * This Action is wired in by AbstractCasMultifactorWebflowConfigurer to put
+ * the resolved provider in the flow scope.
+ *
+ * @author Travis Schmidt
+ * @since 5.3.4
+ */
+public class MfaInitializeAction extends AbstractAction {
+
+    @Override
+    protected Event doExecute(final RequestContext context) throws Exception {
+        final String activeFlow = context.getActiveFlow().getId();
+        final MultifactorAuthenticationProvider provider =
```

Comment:
This does need to move into MFA Utils (I think there is a similar call in there already)

Reply:
I will look into this along with the entire MFA refactor, MFAUtils not currently in the build for this module.